hub.go

```diff
@@ -331,8 +332,22 @@ func hubEntryToEntity(entry *hubEntry) *HubEntity {
 }
 
 // NewHub creates a new Event Hub client for sending and receiving messages
+// NOTE: If the AZURE_ENVIRONMENT variable is set, it will be used to set the ServiceBusEndpointSuffix
+// from the corresponding azure.Environment type at the end of the namespace host string. The default
+// value is azure.PublicCloud. Azure Stack environment is currently not supported.
 func NewHub(namespace, name string, tokenProvider auth.TokenProvider, opts ...HubOption) (*Hub, error) {
-	ns, err := newNamespace(namespaceWithAzureEnvironment(namespace, tokenProvider, azure.PublicCloud))
+	var env azure.Environment
+	switch os.Getenv("AZURE_ENVIRONMENT") {
```

Comment:
Take a look at `azure.EnvironmentFromName()`, it does this heavy lifting for you.

Reply:
Thank you! I wasnt aware of that function. Although, it does return an error and not default to `azure.PublicCloud` like the code currently does. If an error is returned I thought of defaulting to azure.PublicCloud to not break people everyone who has been using it that way until now.